### PR TITLE
Use ledger specifier to write agent key files

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR updates the deployment builder to use ledger identifier property in the keys.json to write the private key files in a deployment setup. 

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes
- [ ] Public-facing documentation has been updated with the changes affected by this PR. Even if the provided contents are not in their final form, all significant information must be included.
- [ ] Any backwards-incompatible/breaking change has been clearly documented in the upgrading document.

### CODE REVIEW

File: docs/advanced_reference/commands/autonomy_deploy.md [hunk 1]
```diff
@@ -114,6 +114,20 @@ autonomy deploy build keys.json -ltm
...
+[
+    {
+        "address": "Solanaaddress",
+        "private_key": "Solanaprivatekey",
+        "ledger": "solana"
+    }
+]
+```
+
+If the keys are generated using `autonomy generate-key` command the `ledger` property will be included automatically.
```
Comment (angrybayblade) [inline docs/advanced_reference/commands/autonomy_deploy.md:129]: This will address the solana service issue @dvilelaf

Comment (dvilelaf) [reply docs/advanced_reference/commands/autonomy_deploy.md:129]: cc @jmoreira-valory

Comment (jmoreira-valory) [reply docs/advanced_reference/commands/autonomy_deploy.md:129]: Does this means that the user has to
1. call generate-keys ethereum --> file 1
2. call generate-keys solana --> file 2
3. manually merge file1 and file2

?

Comment (angrybayblade) [reply docs/advanced_reference/commands/autonomy_deploy.md:129]: You only require 1 keys.json file containing solana keys.

Comment (angrybayblade) [reply docs/advanced_reference/commands/autonomy_deploy.md:129]: When building a deployment you only need a keys.json file containing the keys for the default/primary ledger.

**Review by jmoreira-valory — APPROVED** [review]

**Pull request merged**